### PR TITLE
fix: Improves CIDR validation for access-ip, core-ip and gnb-subnet

### PR DESCRIPTION
**Pull request by Gmerold**
# Description

This PR improves CIDR validation for `access-ip`, `core-ip` and `gnb-subnet`. Before this change it was possible to pass a regular IP address instead of CIDR to the mentioned config options. 

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have bumped the version of the library

### CODE REVIEW

File: tests/unit/test_charm.py [hunk 1]
```diff
@@ -218,6 +219,55 @@ def test_given_invalid_gnbsubnet_config_when_config_changed_then_status_is_block
             "The following configurations are not valid: ['gnb-subnet']"
         )
 
+    def test_given_gnbsubnet_config_is_not_cidr_when_config_changed_then_status_is_blocked(self):
```
Comment (patriciareinoso) [inline tests/unit/test_charm.py:222]: would it be worth it to make these tests parameterized?

Comment (Gmerold) [reply tests/unit/test_charm.py]: Sure, good idea.
Done.

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -218,6 +220,42 @@ def test_given_invalid_gnbsubnet_config_when_config_changed_then_status_is_block
             "The following configurations are not valid: ['gnb-subnet']"
         )
 
+    @parameterized.expand(["192.168.0.1", "192.168.0.1/35", "555.555.555.555/24", "not ip"])
```
Comment (patriciareinoso) [inline tests/unit/test_charm.py:223]: why not `@pytest.mark.parametrize` ?

Comment (Gmerold) [reply tests/unit/test_charm.py]: Hahaha :D Ok :)

**Review by patriciareinoso — APPROVED** [review]

**Pull request merged**